eppoclient/bandits_test.go

```diff
@@ -0,0 +1,134 @@
+package eppoclient
+
+import (
+	"encoding/json"
+	"io"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+)
+
+type banditTest struct {
+	Flag         string
+	DefaultValue string
+	Subjects     []struct {
+		SubjectKey        string
+		SubjectAttributes struct {
+			Numeric     map[string]float64 `json:"numeric_attributes"`
+			Categorical map[string]string  `json:"categorical_attributes"`
+		}
+		Actions []struct {
+			ActionKey             string
+			NumericAttributes     map[string]float64
+			CategoricalAttributes map[string]string
+		}
+		Assignment BanditResult
+	}
+}
+
+func Test_InferContextAttributes(t *testing.T) {
+	attributes := Attributes{
+		"string": "blah",
+		"int":    42,
+		"bool":   true,
+	}
+	contextAttributes := InferContextAttributes(attributes)
+
+	expected := ContextAttributes{
+		Numeric: map[string]float64{
+			"int": 42.0,
+		},
+		Categorical: map[string]string{
+			"string": "blah",
+			"bool":   "true",
+		},
+	}
+
+	assert.Equal(t, expected, contextAttributes)
+}
+
+func Test_bandits_sdkTestData(t *testing.T) {
+	flags := readJsonFile[ufcResponse]("test-data/ufc/bandit-flags-v1.json")
+	bandits := readJsonFile[banditResponse]("test-data/ufc/bandit-models-v1.json")
+	configStore := configurationStore{
+		flags:   flags.Flags,
+		bandits: bandits.Bandits,
+	}
+	logger := new(mockLogger)
+	logger.Mock.On("LogAssignment", mock.Anything).Return()
+	logger.Mock.On("LogBanditAction", mock.Anything).Return()
```

Comment:
we should have _one_ test and test case where we spy on `LogBanditAction` and make sure the right things are being passed as the event.

Reply:
I have added a new test for this